STAR-Delegation/CSR-template/template-schema.cddl

```diff
@@ -0,0 +1,76 @@
+csr-template-schema = {
+  ? keyTypes: [* rsaKeyType / ecKeyType] ; XXX(tho) could IdO specify multiple keyTypes?
+  ? subject: distinguishedName
+  ? extensions: extensions
+}
+
+mandatory-wildcard = "**"
+optional-wildcard = "*"
+wildcard = mandatory-wildcard / optional-wildcard
+
+; regtext matches all text strings but "*" and "**"
+regtext = text .regexp "([^\*].*)|([\*][^\*].*)|([\*][\*].+)"
+
+regtext-or-wildcard = regtext / wildcard
+
+distinguishedName = {
+  ? country: regtext-or-wildcard
+  ? stateOrProvince: regtext-or-wildcard
+  ? locality: regtext-or-wildcard
+  ? organization: regtext-or-wildcard
+  ? organizationalUnit: regtext-or-wildcard
+  ? emailAddress: regtext-or-wildcard
+  ? commonName: regtext-or-wildcard
+}
+
+rsaKeyType = {
+  ? PublicKeyType: "RSA"
+  ? PublicKeyLength: $rsaKeySize / mandatory-wildcard
```

Comment:
No wildcards, and needs to be an integer.

Reply:
Question: how does the IdO let the NDC pick a length?  PublicKeyLength is optional, so maybe omitting it?  But then this would be an exception to the `**` rule in Section.  Or maybe `keyTypes` needs to be treated differently?

Reply:
I think we agree on where this needs to go: `keyTypes` is an array of minimum length 1, and `PublicKeyLength` should be mandatory (and also `PublicKeyType, BTW). Again, `keyTypes` specifically is not a template, it's a constraint (a.k.a. rule) for the public key sent in the CSR.

Reply:
OK, cool.  Sounds good to me.

> `keyTypes` specifically is not a template, it's a constraint (a.k.a. rule) for the public key sent in the CSR.

I think the above should be made crystal clear in the text because it has the potential of generating some confusion.  

`keyTypes` is part of what we call the "template schema" so its "not being a template" is a tad counterintuitive :-)

Reply:
This is what the new text that appears in the current issue means (though maybe not crystal clear): 
> The Subject Public Key Info field of the CSR is generated by the NDC according to the constraints
defined by the "keyTypes" object of the template.